packages/backend/src/peripherals/database/types.ts

```diff
@@ -0,0 +1,34 @@
+export {}
+
+declare module 'knex/types/tables' {
+  interface PositionUpdateRow {
+    position_id: bigint
+    public_key: string
+    collateral_balance: bigint
+    funding_timestamp: bigint
+    balances: JsonB<AssetBalanceJson[]>
+  }
+
+  interface AssetBalanceJson {
+    asset_id: string
+    balance: string
+  }
+
+  interface Tables {
+    position_updates: PositionUpdateRow
+  }
+}
+
+export interface Repository<TRecord> {
```

Comment:
I am unsure if I like this or not, especially with `addOrUpdate`. Let's keep it for now, I am curious to see how this scales for different repositories.